pkg/controller/che/che_cr_validator.go

```diff
@@ -23,7 +23,15 @@ import (
 // - self-contradictory configurations
 // - configurations with which it is impossible to deploy Che
 func ValidateCheCR(checluster *orgv1.CheCluster, isOpenshift bool) error {
-	if !isOpenshift {
+	if isOpenshift {
+		exposureStrategy := checluster.Spec.Server.ServerExposureStrategy
+		exposureType := checluster.Spec.Server.SingleHostExposureType
+		if exposureStrategy != "" && exposureStrategy != "multi-host" {
```

Comment:
isn't `if exposureStrategy == "single=host" && exposureType != "gateway"` the same but simpler?